src/package/packageVersionRetrieve.ts

```diff
@@ -0,0 +1,224 @@
+
+  if (path.isAbsolute(destinationFolder)) {
+    throw messages.createError('sourcesDownloadDirectoryMustBeRelative');
+  }
+
+  const destinationPath = path.join(project.getPath(), destinationFolder);
+  if (!fs.existsSync(destinationPath)) {
+    fs.mkdirSync(destinationPath, { recursive: true });
+  }
+  if (!isPackageDirectoryEffectivelyEmpty(destinationPath)) {
+    throw messages.createError('sourcesDownloadDirectoryNotEmpty');
+  }
+
+  // Get the MetadataZip URL from the MetadataPackageVersion record
+  const subscriberPackageVersionId =
+    project.getPackageIdFromAlias(options.subscriberPackageVersionId) ?? options.subscriberPackageVersionId;
+  const versionInfo: PackagingSObjects.MetadataPackageVersion = (await connection.tooling
+    .sobject('MetadataPackageVersion')
+    .retrieve(subscriberPackageVersionId)) as PackagingSObjects.MetadataPackageVersion;
+
+  if (!versionInfo.MetadataZip) {
+    throw messages.createError('unableToAccessMetadataZip');
+  }
+
+  const responseBase64 = await connection.tooling.request<string>(versionInfo.MetadataZip, {
+    encoding: 'base64',
+  });
+  const buffer = Buffer.from(responseBase64, 'base64');
+
+  let tree = await ZipTreeContainer.create(buffer);
+  let dependencies: string[] = [];
+
+  // 2GP packages declare their dependencies in dependency-ids.json within the outer zip.
+  if (tree.exists('dependency-ids.json')) {
+    interface DependencyIds {
+      ids: string[];
+    }
+    const f = await tree.readFile('dependency-ids.json');
+    const idsObj: DependencyIds = JSON.parse(f.toString()) as DependencyIds;
+    if (idsObj?.ids) {
+      dependencies = idsObj.ids;
+    }
+  }
+
+  // 2GP packages have the package.zip wrapped in an outer zip.
+  if (tree.exists('package.zip')) {
+    tree = await ZipTreeContainer.create(await tree.readFile('package.zip'));
+  }
+
+  const zipComponents = ComponentSet.fromSource({
+    fsPaths: ['.'],
+    tree,
+  })
+    .getSourceComponents()
+    .toArray();
+
+  const result = await new MetadataConverter().convert(zipComponents, 'source', {
+    type: 'directory',
+    outputDirectory: destinationPath,
+    genUniqueDir: false,
+  });
+
+  await attemptToUpdateProjectJson(
+    project,
+    connection,
+    versionInfo.MetadataPackageId,
+    subscriberPackageVersionId,
+    dependencies,
+    destinationFolder
+  );
+
+  return result;
+}
+
+/**
+ * Attempt to update the sfdx-project.json file to add information about the retrieved sources. If this fails for some reason we
+ * print out an error message and return so the user will still see a list of retrieved metadata.
+ *
+ */
+async function attemptToUpdateProjectJson(
+  project: SfProject,
+  connection: Connection,
+  packageId: string,
+  subscriberPackageVersionId: string,
+  dependencyIds: string[],
+  destinationFolder: string
+): Promise<void> {
+  if (process.env.SF_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_VERSION_RETRIEVE) {
```

Comment:
I know this was existing code, but it's not accurate.  If a customer were to set this to `false` they would expect different behavior.